compute_score.py

```diff
@@ -25,22 +24,27 @@
     args = parser.parse_args()
 
     config = None
+    weight_db = None
     if args.configuration is not None:
         with open(args.configuration) as cfg_file:
             config = yaml.full_load(cfg_file)
+            weight_db = get_weights_from_spec(config)
     else:
         if args.benchmark_data_file is None and args.benchmark_data_dir is None:
             parser.print_help(sys.stderr)
             raise ValueError("Invalid command-line arguments. You must specify a config, a data file, or a data dir.")
         with open(SPEC_FILE_DEFAULT) as spec_file:
             spec = yaml.full_load(spec_file)
+            weight_db = get_weights_from_spec(spec)
+
 
     if args.benchmark_data_file is not None:
+        norm = None
         with open(args.benchmark_data_file) as data_file:
             data = json.load(data_file)
-            if config is None:
-                config = generate_bench_cfg(spec, data, TARGET_SCORE_DEFAULT)
-        score = compute_score(config, data)
+            if norm is None:
+                norm = get_benchmark_norms(data)
+        score = compute_score(weight_db, data, TARGET_SCORE_DEFAULT, norm)
```

Comment:
i'm thinking this function has too complicated an API.  Maybe the way to help it is to make a new class on which compute_score is a method.

I'm thinking instead of create these dictionaries and give them confusing names like 'weight_db' or 'spec' or 'config' (not your fault, it's just not possible to come up with a name that explains what the thing does), what if we do this

score_config = TorchBenchScore(args to configure it)
score_config.compute_score(data)

and in the default case, when no spec file is provided, 
from somewhere import TorchBenchmarkScore_v0 //it'd be pre-constructed in 'somewhere'
TorchBenchmarkScore_v0.compute_score(data)


TARGET_SCORE_DEFAULT should be just a default arg to construct the score 

it should be possible to construct a TorchbenchScore() directly from a reference data, without a spec, i think.  
from somewhere import TorchBenchScore
score_config = TorchBenchScore.from_reference_data(ref_data)
score_config.compute_score()

I wonder if it's really important to have the weight_db or spec as an exposed thing?  If the goal is to enable filtering out models (assume the reference data has too many models, but you don't care about all of them), maybe just pass a list of excludes?
score_config = TorchBenchScore.from_reference_data(ref_data, exclude_models=None, exclude_tests=None, exclude_device=None)...
score_config.compute_score()